File: hyper_rendering/src/resource_handle.rs
/*
 * Copyright (c) 2023, SkillerRaptor
 *
 * SPDX-License-Identifier: MIT
 */

#[repr(transparent)]
#[derive(Clone, Copy, Debug)]
pub(crate) struct ResourceHandle(u32);

impl ResourceHandle {
    pub(crate) fn new(index: u32) -> Self {
        Self(index)
    }

    pub(crate) fn index(&self) -> u32 {
        self.0 & ((1 << 24) - 1)
    }
}

File: hyper_rhi/src/hyper_rhi/resource_handle.cpp
/*
 * Copyright (c) 2024, SkillerRaptor
 *
 * SPDX-License-Identifier: MIT
 */

#include "hyper_rhi/resource_handle.hpp"

namespace hyper_rhi
{
    ResourceHandle::ResourceHandle(const uint32_t handle)
        : m_handle(handle)
    {
    }

    uint32_t ResourceHandle::handle() const
    {
        return m_handle;
    }
} // namespace hyper_rhi